libwhisper/WhisperDB.cpp

```diff
@@ -134,3 +135,54 @@ void WhisperDB::loadAll(std::map<h256, Envelope>& o_dst)
 	}
 }
 
+void WhisperDB::save(h256 const& _key, Envelope const& _e)
+{
+	try
+	{
+		RLPStream meta;
+		streamMetaInfo(meta, _e);
+
+		RLPStream msg;
+		_e.streamRLP(msg);
+
+		RLPStream res(2);
+		res.appendRaw(meta.out());
+		res.appendRaw(msg.out());
+
+		bytes b;
+		res.swapOut(b);
+		insert(_key, b);
+	}
+	catch(RLPException const& ex)
+	{
+		cwarn << "RLPException in WhisperDB::save():" << ex.what();
```

Comment:
For boost exceptions diagnostic_information or current_exception_diagnostic_information should be used.